app/src/main/java/com/github/epfl/meili/messages/NewMessageActivity.kt

```diff
@@ -0,0 +1,74 @@
+package com.github.epfl.meili
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.widget.TextView
+import androidx.recyclerview.widget.RecyclerView
+import com.github.epfl.meili.models.User
+import com.google.firebase.database.DataSnapshot
+import com.google.firebase.database.DatabaseError
+import com.google.firebase.database.ValueEventListener
+import com.google.firebase.database.ktx.database
+import com.google.firebase.ktx.Firebase
+import com.xwray.groupie.GroupAdapter
+import com.xwray.groupie.GroupieViewHolder
+import com.xwray.groupie.Item
+
+class NewMessageActivity : AppCompatActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContentView(R.layout.activity_new_message)
+
+        supportActionBar?.title = "Select User"
+
+
+        fetchUsers()
+    }
+
+    companion object {
+        val USER_KEY = "USERK_KEY"
+    }
+
+    private fun fetchUsers() {
```

Comment:
I think it would be good for modularisation to have this in an auxiliary ViewModel class. But since this is already a huge PR maybe you can add in the Backlog a task that asks for refactoring of this code and leave it for later.

Reply:
I am saying this because I believe that ideally the Activities should only do UI related actions and not the actions that correspond to the model as fetching users from database.

Reply:
Totally agree! Refactoring the code to comply with the MVCC model will be my top priority for the next sprint.